middleware/faq.rst

```diff
@@ -436,3 +436,77 @@ What do I do if a query method/command or pipetask graph generation is slow?
+      refcats/DM-28636                                   RUN
+      skymaps                                            RUN
+    u/jbosch/DM-30649/20210614T191615Z                   RUN
+
+The RUN collections that directly hold the datasets are what we want to remove in order to free up space, but we have to start by deleting the `~CollectionType.CHAINED` collections that hold them first:
+
+.. code:: sh
+
+    $ butler remove-collection /repo/main u/jbosch/DM-30649 --no-confirm
+
+You can drop the ``--no-confirm`` option if you like.
+If you're only deleting one collection at a time, it doesn't tell you anything new.
+
+If you don't want to remove the `~CollectionType.CHAINED` collection - you just want to remove the `~CollectionType.RUN` collection from it - you can instead do
+
+    $ butler collection-chain /repo/main --remove u/jbosch/DM-30649 u/jbosch/DM-20210614T191615Z
+
+Or, if you know the `~CollectionType.RUN` is the first one in the chain,
+
+    $ butler collection-chain /repo/main --pop u/jbosch/DM-30649
+
+In any case, once the `~CollectionType.CHAINED` collection is out of the way, we can delete the `~CollectionType.RUN` collections that start with the same prefix using a glob pattern:
+
+.. code:: sh
+
+    $ butler remove-runs /repo/main u/jbosch/DM-30649/*
+    The following RUN collections will be removed:
+    u/jbosch/DM-30649/20210614T191615Z
+    The following datasets will be removed:
+    calexp(18222), calexpBackground(18222), calexp_camera(168), calibrate_config(1), calibrate_metadata(18222), characterizeImage_config(1), characterizeImage_metadata(18231), consolidateSourceTable_config(1), consolidateVisitSummary_config(1), consolidateVisitSummary_metadata(168), fgcmBuildStarsTable_config(1), fgcmFitCycle_config(1), fgcmOutputProducts_config(1), icExp(18231), icExpBackground(18231), icSrc(18231), icSrc_schema(1), isr_config(1), isr_metadata(18232), postISRCCD(18232), skyCorr(17304), skyCorr_config(1), skyCorr_metadata(168), source(18222), src(18222), srcMatch(18222), srcMatchFull(18222), src_schema(1), transformSourceTable_config(1), visitSummary(168), writeSourceTable_config(1), writeSourceTable_metadata(18222)
+    Continue? [y/N]: y
+    Removed collections
+
+Here we've left the default confirmation behavior on because we used a glob, just to be safe.
+You can write one or more full RUN collection names explicitly, too, and that's what you'll need to do if you didn't follow the naming convention well enough for a glob to work.
+
+Removing `~CollectionType.RUN` collections always removes the files within them, but it does not remove the directory structure, because in the presence of arbitrary path templates (including any that may have been used in the past) and possible concurrent writes, it's difficult for the butler to recognize efficiently when a directory will end up empty.
+You're welcome to delete empty directories on your own after using ``remove-runs``.
```

Comment:
The first time I read this sentence, I thought it was telling me that I could delete empty directories using `remove-runs`. Would something like this instead read well?
```
After the `remove-runs` command has successfully completed, you're welcome to delete empty directories on your own. 
Empty directories can be found in `REPO/COLLECTION`.
```

Reply:
PS - is it necessary/good-practice to delete empty directories, i.e., does this have a big impact on how the datastore will operate, or is it safe to leave these empty directories in-place? After some initial entirely self-created disasters in manually deleting empty directories, I've now leaned towards leaving them be. 

If it is ~okay to leave them be, perhaps it might be good to add a note here saying as much?

Reply:
The datastore doesn't care about the empty directories. There is no requirement to clean them up. In general there's no need for people to care how the datastore is laid out (which will be more opaque with an object store).

Reply:
I think it's good to add a comment stating as much (that it's okay to leave empty directories in place), to help provide guidance to those that may be wondering.